src/csharp/Record/Playback.cs

```diff
@@ -0,0 +1,786 @@
+            }
+        }
+
+        /// <summary>
+        /// Gets get the number of tracks in a playback file.
+        /// </summary>
+        public int TrackCount
+        {
+            get
+            {
+                lock (this)
+                {
+                    if (this.disposedValue)
+                    {
+                        throw new ObjectDisposedException(nameof(Playback));
+                    }
+
+                    return checked((int)NativeMethods.k4a_playback_get_track_count(this.handle));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the length of the recording in microseconds.
+        /// </summary>
+        /// <remarks>
+        /// The recording length, calculated as the difference between the first and last timestamp in the file.
+        ///
+        /// The recording length may be longer than an individual track if, for example, the IMU continues to run after the last
+        /// color image is recorded.
+        /// </remarks>
+        public TimeSpan RecordingLength
+        {
+            get
+            {
+                if (this.disposedValue)
+                {
+                    throw new ObjectDisposedException(nameof(Playback));
+                }
+
+                long length = checked((long)NativeMethods.k4a_playback_get_recording_length_usec(this.handle));
+                return TimeSpan.FromTicks(length * 10);
+            }
+        }
+
+        /// <summary>
+        /// Opens an existing recording file for reading.
+        /// </summary>
+        /// <param name="path">Filesystem path of the existing recording.</param>
+        /// <returns>An object representing the file for playback.</returns>
+        public static Playback Open(string path)
+        {
+            NativeMethods.k4a_playback_t handle = null;
+
+            AzureKinectOpenPlaybackException.ThrowIfNotSuccess(() => NativeMethods.k4a_playback_open(path, out handle));
+
+            return new Playback(handle);
+        }
+
+        /// <summary>
+        /// Get the raw calibration blob for the Azure Kinect device used during recording.
+        /// </summary>
+        /// <returns>The raw calibration may not exist if the device was not specified during recording.</returns>
+        public byte[] GetRawCalibration()
+        {
+            lock (this)
+            {
+                if (this.disposedValue)
+                {
+                    throw new ObjectDisposedException(nameof(Playback));
+                }
+
+                // Determine the required calibration size
+                UIntPtr size = new UIntPtr(0);
+                if (NativeMethods.k4a_playback_get_raw_calibration(this.handle, null, ref size) != NativeMethods.k4a_buffer_result_t.K4A_BUFFER_RESULT_TOO_SMALL)
+                {
+                    throw new AzureKinectGetRawCalibrationException($"Unexpected result calling {nameof(NativeMethods.k4a_playback_get_raw_calibration)}");
```

Comment:
If I understand the code correctly, this will throw if there is no calibration, however the exception will just say unexpected result. Can we add logging in here? Or somehow return a better result for when the calibration doesn't exist?